### PR TITLE
U+0085 not recognised as a newline character in CSharpCodeGenerator.cs (#1740)

### PR DESCRIPTION
**Pull request by hamarb123**
U+0085 is not recognised as a newline character in CSharpCodeGenerator.cs, this fixes that.

This an issue is because it is recognised as a newline character in C# ([see here](https://github.com/dotnet/csharplang/blob/master/spec/lexical-structure.md)), which cannot be in literal strings.

Specifically:
```antlr
new_line_character
    : '<Carriage return character (U+000D)>'
    | '<Line feed character (U+000A)>'
    | '<Next line character (U+0085)>'
    | '<Line separator character (U+2028)>'
    | '<Paragraph separator character (U+2029)>'
    ;
```

and

```antlr
regular_string_literal
    : '"' regular_string_literal_character* '"'
    ;

regular_string_literal_character
    : single_regular_string_literal_character
    | simple_escape_sequence
    | hexadecimal_escape_sequence
    | unicode_escape_sequence
    ;

single_regular_string_literal_character
    : '<Any character except " (U+0022), \\ (U+005C), and new_line_character>'
    ;
```

If the CSharpCodeGenerator class was used to make code with a string in it that has [U+0085](http://www.unicode-symbol.com/u/0085.html) in it, it would generate invalid C# code.

I have also added a test which ensures that all of the newline characters (other than \n and \r) are generated properly.

Fix for #1740

### CODE REVIEW

File: src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -125,6 +125,7 @@ private string QuoteSnippetStringCStyle(string value)
                         break;
                     case '\u2028':
                     case '\u2029':
+                    case '\u0085':
```
Comment (hamarb123) [inline src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: This is a newline character, see #1740 and #1853.

File: src/libraries/System.CodeDom/tests/System/CodeDom/Compiler/CSharpCodeGenerationTests.cs [hunk 2]
```diff
@@ -1066,7 +1066,7 @@ public void CharEncoding()
 
             AssertEqual(main,
                  "public static void Main() { " +
-                 "     System.Console.WriteLine(\"\u1234 \u4567 \uABCD \\r \\n \\t \\\\ \\\" \\' \\0 \\u2028 \\u2029 \u0084 \u0085 \U00010F00\"); " +
+                 "     System.Console.WriteLine(\"\u1234 \u4567 \uABCD \\r \\n \\t \\\\ \\\" \\' \\0 \\u2028 \\u2029 \u0084 \\u0085 \U00010F00\"); " +
```
Comment (hamarb123) [inline src/libraries/System.CodeDom/tests/System/CodeDom/Compiler/CSharpCodeGenerationTests.cs]: This should not be unescaped as that would be illegal C# code, see #1740 and #1853.

**Comment by hamarb123** [comment]
It seems like the test is failing on 2 platforms: Windows_NT netfx x86 Release, and Windows_NT x64.
I imagine the reasoning behind the netfx failure is because it must still have the old code, but can anyone tell me what the other one is, its error messages don't make sense to me.
Also, what should I do to fix the netfx one, should I just revert the tests for the netfx platform, or is there something else that I should do?

**Comment by hamarb123** [comment]
This change will be disabled for .NET Framework, and Microsoft can add it if they wish, see #1913

**Comment by hamarb123** [comment]
As all of the checks have now passed, this is now ready for review.

**Review by cston — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
LGTM. Thanks.

**Comment by cston** [comment]
Thanks @hamarb123!

**Pull request merged**